src/components/Form/FieldControllers/SwitchController/index.tsx

```diff
@@ -0,0 +1,48 @@
+import { Switch } from "@mui/material";
+import { Control, Controller, FieldValues, Path } from "react-hook-form";
+
+import Box from "@/components/Box";
+import Typography from "@/components/Typography";
+
+type Props<T extends FieldValues> = {
+  [key: string]: any;
+  control: Control<T>;
+  label: string;
+  name: Path<T>;
+  required?: boolean;
+};
+
+const SwitchController = <T extends FieldValues>({
+  name,
+  control,
+  required,
+  label,
+  ...rest
+}: Props<T>) => (
+  <Controller
+    name={name}
+    control={control}
+    rules={{
+      required: required && "This field is required",
+    }}
+    render={({ field: { onChange, value }, fieldState: { error } }) => {
+      return (
+        <Box display="flex" alignItems="center" lineHeight={1}>
```

Comment:
Please consider creating component level styles files, in the form of css-in-js, and passing it to MUI components' `sx` property. We shouldn't have inline styling for components of elements. Example:

```suggestion
        ....
        const styles = {
           display: "flex",
           alignItems: "center",
           lineHeight: 1
        }
        <Box sx={styles.box}>
```